src/game/client/water/w3dwatertracks.cpp

```diff
@@ -325,9 +329,213 @@ int WaterTracksObj::Update(int ms_elapsed)
 
 void Test_Water_Update()
 {
-    // todo needs InGameUI
-#ifdef GAME_DLL
-    Call_Function<void>(PICK_ADDRESS(0x007A22C0, 0x0064DC2D));
+#ifdef BUILD_WITH_D3D8
+    static bool pause_waves = false;
+    static POINT mouse_anchor;
+    static Coord3D terrain_point_start;
+    static Coord3D terrain_point_end;
+    static WaterTracksObj *track = nullptr;
+    static WaterTracksObj *track2 = nullptr;
+    static int track_edit_mode = 0;
+    static bool have_start = false;
+    static bool have_end = false;
+    static bool do_init = true;
+    static WaveType current_wave_type = WAVE_TYPE_OCEAN;
+    static bool track_edit_mode_reset = true;
+    static bool add_point_reset = true;
+    static bool delete_track_reset = true;
+    static bool save_tracks_reset = true;
+    static bool load_tracks_reset = true;
+    static bool change_type_reset = true;
+
+    pause_waves = false;
+
+    if (do_init == true) {
+        do_init = false;
+    }
+
+    if ((GetAsyncKeyState(VK_F5) & 0x8001) != 0) {
+        if (track_edit_mode_reset == true) {
+            if (track_edit_mode != 0) {
+                Utf16String str;
+                str.Format(L"Leaving Water Track Edit Mode");
+                g_theInGameUI->Message(str);
+            } else {
+                Utf16String str;
+                str.Format(L"Entering Water Track Edit Mode");
+                g_theInGameUI->Message(str);
+                str.Format(L"Wave Type: %hs", s_waveTypeInfo[current_wave_type].wave_name);
+                g_theInGameUI->Message(str);
+            }
+
+            track_edit_mode ^= 1u;

```

Comment:
track_edit_mode can be made a bool its only ever 1 or 0.

Reply:
fixed.